### PR TITLE
AWS Access Analyzer Fetch

### PR DESCRIPTION
**Pull request by nericksen**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
Fetch was not built out on AWS Access Analyzer Integration.  This PR adds the functionality to enable fetching of Findings from Access Analyzer.

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [x] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py [hunk 1]
```diff
@@ -309,24 +316,81 @@ def test_function():
...
+        last_fetch = date_to_timestamp(delta)
+
+    incidents: list = []
+    dArgs = {}
+    dArgs['roleArn'] = dParams.get('roleArn')
+    dArgs['region'] = dParams.get('region')
+    dArgs['roleSessionName'] = dParams.get('roleSessionName')
+    dArgs['roleSessionDuration'] = dParams.get('roleSessionDuration')
+    dArgs['analyzerArn'] = dParams.get('analyzerArn')
+    dArgs['maxResults'] = 25
```
Comment (liorblob) [inline Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py:337]: It's better to initialize params in the `main` function

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py [hunk 2]
```diff
@@ -309,24 +316,81 @@ def test_function():
...
+
+    nextToken = None
+    incidents = []
+    tmp_last_fetch = last_fetch
+
+    while True:
+
+        if nextToken:
+            dArgs['nextToken'] = nextToken
+        raw_incidents = list_findings_command(dArgs)
```
Comment (liorblob) [inline Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py:347]: better separate the findings function to a request function and a command function

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py [hunk 3]
```diff
@@ -309,24 +316,81 @@ def test_function():
...
-        get_analyzed_resource_command(demisto.args())
-    elif demisto.command() == 'aws-access-analyzer-get-finding':
-        get_finding_command(demisto.args())
-    elif demisto.command() == 'aws-access-analyzer-start-resource-scan':
-        start_resource_scan_command(demisto.args())
-    elif demisto.command() == 'aws-access-analyzer-update-findings':
-        update_findings_command(demisto.args())
-
-except Exception as e:
-    return_error(f"Error has occured in AWS Access Analyzer Integration: {str(e)}")
```
Comment (liorblob) [inline Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py:332]: why was this removed?

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py [hunk 4]
```diff
@@ -309,24 +316,81 @@ def test_function():
...
+    result = test_function()
+elif demisto.command() == 'fetch-incidents':
+    incidents = fetch_incidents(demisto.getLastRun())
+    demisto.incidents(incidents)
+elif demisto.command() == 'aws-access-analyzer-list-analyzers':
+    list_analyzers_command(demisto.args())
+elif demisto.command() == 'aws-access-analyzer-list-analyzed-resource':
+    list_analyzed_resource_command(demisto.args())
+elif demisto.command() == 'aws-access-analyzer-list-findings':
+    data = list_findings_command(demisto.args())
```
Comment (liorblob) [inline Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py:385]: Better separate this so the command function stays as is as stated above

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.yml [hunk 5]
```diff
@@ -61,6 +61,11 @@ configuration:
   name: proxy
   required: false
   type: 8
+- display: Fetch Analyzer ARN
```
Comment (liorblob) [inline Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.yml:64]: Please add the `additionalinfo` property to explain this parameter.

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py (hunk 3 above)
Comment (nericksen) [reply Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py:332]: This try and catch is around the entire block of code that is executed for this integration, so it made it difficult to debug errors as they are always reported as coming from the exception block. This is opposed to reporting the exact line of code that threw the error.  Might be an improvement to consider wrapping each API call in a try catch instead of 1 global one.  Essentially I removed it to make debugging easier. I can add this back if you think it is best here.

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py (hunk 1 above)
Comment (nericksen) [reply Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py:337]: It seems this integration does not have a main function, thoughts on leaving this parameter initialization in the fetch function until the entire integration is rewritten to comply with all our current standards?

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.yml (hunk 5 above)
Comment (nericksen) [reply Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.yml:64]: will do

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py (hunk 1 above)
Comment (liorblob) [reply Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py:337]: sounds good to me

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py (hunk 3 above)
Comment (liorblob) [reply Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py:332]: It is best as it was so the user does not get the traceback in the error when they run the command.

Comment (nericksen) [reply Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py:332]: 👍

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py (hunk 4 above)
Comment (nericksen) [reply Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py]: reverted the changes back so this is uniform with the rest of the commands

File: Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py (hunk 2 above)
Comment (nericksen) [reply Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py]: Created a new function `get_findings_command` to use with fetch command.

**Comment by liorblob** [comment]
@nericksen looks like you have some flake8 issues, you can have a look on the Run Unit Testing and Lint section of the checks.
Also, please update release notes by running `demisto-sdk update-release-notes -p AWS-AccessAnalyzer -u revision`.

**Comment by nericksen** [comment]
@liorblob linting should now be fixed and I have now added release notes.  It doesn't seem like there are any unit tests yet for this integration.  Did you want me to look into adding tests for this?

**Comment by liorblob** [comment]
@nericksen since this pack does not have any tests, I don't think they are required here. But if you're willing to add some tests for the fetch you're more than welcome.

**Comment by nericksen** [comment]
ok I will defer unit testing for now then.  I will need to do some further reading on unit testing before implementing.  Will push in a separate PR in order to not hold this one up if it isn't necessary.
